e2e/typescript/src/assets/fixture-reset.ts

```diff
@@ -0,0 +1,321 @@
+import { GriffelResetStyle } from '@griffel/core';
+
+function assertType(style: GriffelResetStyle): GriffelResetStyle {
+  return style;
+}
+
+// Animation
+assertType({ animationName: 'foo' });
+
+// Basic styles
+//
+
+assertType({ flex: 0 });
+assertType({ flex: 1 });
+assertType({ zIndex: 0 });
+assertType({ zIndex: 1 });
+
+assertType({ padding: '5px' });
+assertType({ paddingLeft: '5px' });
+assertType({ color: 'beige' });
+
+// CSS variables
+//
+
+assertType({ fontWeight: 'var(--foo)' });
+assertType({ flexShrink: 'var(--bar)' });
+assertType({ opacity: 'var(--baz)' });
+assertType({ zIndex: 'var(--qux)' });
+
+assertType({ '--color': 'red' });
+
+// Mixins
+//
+
+assertType({
+  ...{ padding: '5px' },
+  ...{ paddingLeft: '5px' },
+  ...{ color: 'red' },
+});
+
+// Strict selector defined via "CSS.Pseudos"
+//
+
+assertType({
+  ':hover': { flexShrink: 0 },
+  ':focus': { flexShrink: 'initial' },
+  ':active': { zIndex: 0 },
+  ':visited': { zIndex: 1 },
+});
+
+assertType({
+  ':hover': { fontWeight: 'var(--foo)' },
+  ':focus': { flexShrink: 'var(--bar)' },
+  ':active': { opacity: 'var(--bar)' },
+  ':visited': { zIndex: 'var(--qux)' },
+});
+
+assertType({
+  ':hover': { color: 'beige' },
+  ':active': { padding: '5px' },
+  ':focus': { paddingLeft: '5px' },
+});
+
+assertType({
+  ':hover': { '--color': 'red' },
+});
+
+// Custom selectors
+//
+
+assertType({
+  ':hover:focus': { flexShrink: 0 },
+  ':hover:active': { flexShrink: 'initial' },
+  ':hover:visited': { zIndex: 0 },
+  ':hover:focus-visible': { zIndex: 1 },
+});
+
+assertType({
+  ':link:hover': { fontWeight: 'var(--foo)' },
+  ':link:focus': { flexShrink: 'var(--bar)' },
+  ':link:active': { opacity: 'var(--bar)' },
+  ':link:visited': { zIndex: 'var(--qux)' },
+});
+
+assertType({
+  '.bar': { color: 'beige' },
+  '.foo': { padding: '5px' },
+  '.qux': { paddingLeft: '5px' },
+});
+
+assertType({
+  '.bar': { '--color': 'red' },
+});
+
+// Nested custom selectors
+//
+
+assertType({
+  '.foo': {
+    '.bar': { flexShrink: 0 },
+    '.baz': { flexShrink: 'initial' },
+    '.qux': { opacity: 0 },
+    '.fred': { zIndex: 0 },
+    '.thud': { zIndex: 1 },
+  },
+  '.bar': {
+    '.baz': { color: 'beige' },
+    '.qux': { paddingLeft: '5px' },
+  },
+  '.baz': {
+    '.qux': {
+      '--color': 'red',
+    },
+  },
+  '.qux': {
+    '.bar': { flexShrink: 'var(--bar)' },
+    '.baz': { opacity: 'var(--baz)' },
+  },
+});
+
+// Invalid values
+//
+
+assertType({
+  // @ts-expect-error "outline-box" is an invalid value for "box-sizing"
+  boxSizing: 'outline-box',
+});
+assertType({
+  // @ts-expect-error "1" is invalid value for "overflowX"
+  overflowX: '1',
+});
+assertType({
+  // @ts-expect-error "paddingLeft" cannot be numeric value
+  paddingLeft: 5,
+});
+assertType({
+  // @ts-expect-error "0" is invalid value for "color"
+  color: 0,
+});
+
+assertType({
+  // @ts-expect-error type check still fails on outline-box, not on any other line
```

Comment:
nit: I would like to see the test for the two lines swapped as well.
Is it expected to fail on the first property in the object (first line) or on the invalid `boxSizing`?

Reply:
Added in a487682.

> Is it expected to fail on the first property in the object (first line) or on the invalid boxSizing?

It fails on invalid `boxSizing`